compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -371,6 +371,7 @@ class OMR_EXTENSIBLE CodeGenerator
    TR_HasRandomGenerator randomizer;
 
    virtual bool supportsAtomicAdd() {return false;}
+   virtual bool supportsTM()    {return false;}
```

Comment:
Hi Daryl, I make it virtual because I hope to disable TM for all platforms except X. (because only X has TM evaluator now.) I think I should change a name for supportsTM(), perhaps change to hasTMEvaluator(). Sorry it's very confusing similar names used for different checks. a) supportsTM() is used for checking if this platform has TM evaluator in code. Early this week we found some jitbuilder testcases failed on P and Z, so Mark asked me to add an assertion in the beginning of the jitbulder service. That's why now I have virtual supportsTM() returning false by default and supportsTM() in X/codgen is true b) getSupportsTM() is used for detecting CPUID to tell if the processor supports TM.